packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import { pick } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { isUnmodifiedDefaultBlock } from '@wordpress/blocks';
+import { _n } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * @typedef WPInserterConfig
+ *
+ * @property {string=}   rootClientId        Inserter Root Client ID.
+ * @property {string=} clientId            Inserter Client ID.
+ * @property {boolean}   isAppender          Whether the inserter is an appender or not.
+ * @property {boolean}   selectBlockOnInsert Whether the block should be selected on insert.
+ */
+
+/**
+ * Returns the insertion point state given the inserter config.
+ *
+ * @param {WPInserterConfig} config Inserter Config.
+ * @return {Array} Insertion Point State (rootClientID, onInsertBlocks and onToggle).
+ */
+function useInsertionPoint( {
+	rootClientId,
+	clientId,
+	isAppender,
+	selectBlockOnInsert,
+} ) {
+	const {
+		destinationRootClientId,
+		getSelectedBlock,
+		getBlockIndex,
+		getBlockSelectionEnd,
+		getBlockOrder,
+	} = useSelect(
+		( select ) => {
+			const {
+				getSettings,
+				getBlockRootClientId,
+				getBlockSelectionEnd: _getBlockSelectionEnd,
+			} = select( 'core/block-editor' );
+
+			let destRootClientId = rootClientId;
+			if ( ! destRootClientId && ! clientId && ! isAppender ) {
+				const end = _getBlockSelectionEnd();
+				if ( end ) {
+					destRootClientId = getBlockRootClientId( end ) || undefined;
```

Comment:
More semantically accurate, I think:
```suggestion
					destRootClientId = getBlockRootClientId( end ) ?? undefined;
```